drivers/counter/counter_nrfx_timer.c

```diff
@@ -89,6 +90,7 @@ static uint32_t read(const struct device *dev)
 
 	nrf_timer_task_trigger(timer,
 			       nrf_timer_capture_task_get(COUNTER_READ_CC));
+	barrier_dmem_fence_full();
```

Comment:
shouldn't this be using `nrf_timer_task_trigger` and `nrfy_timer_task_trigger`? Isn't this the whole idea of nrfy?

Reply:
nrfy was created with assumption that any periperhal access from vpr core can be reordered but this is not the case anymore. Reordering can only occur between different type of operations (read or write). In this driver i've identified 2 places.
When i tried to convert driver to use nrfy_timer then it gets +200 bytes of code. Given that vpr cores have very limited memory imo it's better to have barriers only where necessary.

Reply:
so, is gonna nrfy be removed?

Reply:
we will discuss it internally but seems that there is no point of using barriers everywhere if they are needed only in very limited number of places.